pkg/services/tree/replicator.go

```diff
@@ -130,6 +130,71 @@ func (s *Service) pushToQueue(cid cidSDK.ID, treeID string, op *pilorama.LogMove
 	}
 }
 
+func (s *Service) synchronize(ctx context.Context) {
```

Comment:
Let's put it in `sync.go`? Also it would be nice to parametrize it by `CID`. When a node first appears in container without shutdown is also a valid use-case.